openedx/MITx-edx-integration-tutor.md

```diff
@@ -0,0 +1,217 @@
+
+
+Tutor Setup, Part Two
+---------------------
+
+Note that some of these steps require editing the main configuration files for the production instance (which is also used for a local deployment). Most of the settings that need to be adjusted to get integration working are overridden by the default Tutor configuration, so you can't update them by setting `config.yml`. If you're using the development Tutor build, you'll likely need to edit `development.py` rather than `production.py` as necessary.
+
+These steps will also disable the AuthN SSO MFE, so from here on you'll get normal edX authentication screens (if you're not being bounced to MITx Application).
+
+
+1. Get the gateway IP of the `mitxApplication_default` Docker network:
+
+        docker network inspect mitxpro_default | grep Gateway
+
+2. Log into to edX using your superuser account, and make sure your session stays open. Sessions are pretty long-lived so this just means not closing the browser that you started the session in. (Part of this process will involve mostly breaking authentication so it's important that you are able to get into the admin.)
+3. Stop Tutor: `tutor local stop`
+4. Change into the configuration root for Tutor:
+
+        cd "$(tutor config printroot)"
+
+5. Create a `env/build/openedx/requirements/private.txt` with the required extensions:
+
+        social-auth-mitxpro
+        openedx-companion-auth
+
+6. Edit the `env/apps/openedx/config/lms.env.yml` file and add:
+
+        FEATURES:
+            ALLOW_PUBLIC_ACCOUNT_CREATION: true
+            ENABLE_COMBINED_LOGIN_REGISTRATION: true
+            ENABLE_THIRD_PARTY_AUTH: true
+            ENABLE_OAUTH2_PROVIDER: true
+            SKIP_EMAIL_VALIDATION: true
+
+   The `FEATURES` block (should be at the top).
+
+7. Edit the `env/apps/openedx/settings/lms/production.py` and/or `env/apps/openedx/settings/lms/development.py` settings file. (The former is used by a local instance, where the latter is used by both dev and nightly instances.)
+
+   * Add to the end of the file:
+
+      * `THIRD_PARTY_AUTH_BACKENDS = ['social_auth_mitxpro.backends.MITxProOAuth2']`
+      * `REGISTRATION_EXTRA_FIELDS["country"] = "hidden"`
+      * `AUTHENTICATION_BACKENDS.append('social_auth_mitxpro.backends.MITxProOAuth2')`
+      * `SOCIAL_AUTH_OAUTH_SECRETS = {"mitxpro-oauth2": <MIT_Client_Secret> }` - Client Secret that you just copied after `configure_instance` management command
+      * `IDA_LOGOUT_URI_LIST.append('http://{Domain}:{PORT}/logout/')` - there's an existing one of these around like 300 in production.py too.
+
+    * Find and update:
+
+      * `FEATURES['ENABLE_AUTHN_MICROFRONTEND'] = False` (defaults to True)
+      * `REGISTRATION_EXTRA_FIELDS["terms_of_service"] = "hidden"` (defaults to required)
+
```

Comment:
I think we can add all these configurations in `private.py`. Let's make is similar to what we have done in devstack.py

Reply:
using tutor does not provide us access to `edx-platform` unless we mount it.

Reply:
Yes, you are correct, we can leave this as is